src/deviceCommunication/commDevice.h

```diff
@@ -0,0 +1,154 @@
+/******************************************************************************
+ * Copyright (C) 2022 by Gschwind, Weber, Schoch, Niederberger                *
+ *                                                                            *
+ * This file is part of linescaleGUI.                                         *
+ *                                                                            *
+ * LinescaleGUI is free software: you can redistribute it and/or modify       *
+ * it under the terms of the GNU General Public License as published by       *
+ * the Free Software Foundation, either version 3 of the License, or          *
+ * (at your option) any later version.                                        *
+ *                                                                            *
+ * LinescaleGUI is distributed in the hope that it will be useful,            *
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the               *
+ * GNU General Public License for more details.                               *
+ *                                                                            *
+ * You should have received a copy of the GNU General Public License          *
+ * along with linescaleGUI. If not, see <http://www.gnu.org/licenses/>.       *
+ ******************************************************************************/
+/**
+ * @file commDevice.h
+ * @authors Gschwind, Weber, Schoch, Niederberger
+ *
+ * @brief Abstract base class for each type of connection
+ *
+ */
+
+#pragma once
+#ifndef COMMDEVICE_H_
+#define COMMDEVICE_H_
+
+#include <QDebug>
+#include <QObject>
+
+namespace comm {
+
+/**
+ * @brief Enum to describe the type used for a specific connection
+ *
+ */
+enum class CONNTYPE {
+    BLE,  ///< Bluetooth low energy
+    USB,  ///< Serial port via USB-mini
+};
+
+/**
+ * @brief Struct to identify a device
+ * Used to connect to one device.
+ *
+ */
+struct DEVICEINFO {
+    CONNTYPE type;  ///< Type of connection
+    QString ID;     ///< Identifier of a given connection; e.g. COM101
+    int baudRate;   ///< Baudrate, used by USB connection
+};
+
+/**
+ * @brief Base class for every connection
+ *
+ */
+class CommDevice : public QObject {
+    Q_OBJECT
+
+   public:
+    /**
+     * @brief Connect device to host
+     * Implemented in USB or BLE class
+     *
+     * @return true Connection established successfully
+     * @return false Connection failed
+     */
+    virtual bool connectDevice() { return false; };
+
+    /**
+     * @brief Disconnect device from host
+     * Implemented in USB or BLE class
+     *
+     */
+    virtual void disconnectDevice(){};
+
+    /**
+     * @brief Send date to the connected device
+     *
+     * @param data HEX values as a QByteArray to send
+     */
+    virtual void sendData(QByteArray data) { qDebug() << data; };
```

Comment:
Probably not a good idea to copy the data since [`QSerialPort::write`](https://doc.qt.io/qt-5/qiodevice.html#write-2) also takes it by `const &`.
```suggestion
    virtual void sendData(const QByteArray& data) { qDebug() << data; };
```